cmd/gcp-controller-manager/csr_approver.go

```diff
@@ -628,13 +628,18 @@ func clusterHasInstance(ctx *controllerContext, instanceZone string, instanceID
 
 func groupHasInstance(ctx *controllerContext, groupLocation, groupName string, instanceID uint64) (bool, error) {
 	recordMetric := csrmetrics.OutboundRPCStartRecorder("compute.InstanceGroupManagersService.ListManagedInstances")
-	instances, err := compute.NewInstanceGroupManagersService(ctx.gcpCfg.Compute).ListManagedInstances(ctx.gcpCfg.ProjectID, groupLocation, groupName).Do()
+	managedInstances := []*compute.ManagedInstance{}
```

Comment:
the accumulator change looks like it will work, but if we're looking for a single instance, is there a way to look up just that one rather than accumulating all, then iterating to find the one that matches?

Reply:
+1, at a minimum we could use a filter rather than accumulater. It would prevent us walking the list twice and not need as much memory.

Reply:
Done.